### PR TITLE
[16.0][OU-ADD] stock

### PR DESCRIPTION
**Pull request by marielejeune**
This Pull Request cherry-picks commits from https://github.com/OCA/OpenUpgrade/pull/3966 where a huge work was already carried (thanks to hoangtiendung070797 for that) but where the last review comments are waiting since a long time.

More precisely, these 2 changes were performed, following the comments on the other PR:
* Method `rename_fields` already performs the column renaming, so calling `rename_columns` is useless.
* Fixed SQL query in `_compute_stock_location_replenish_location`.

### CODE REVIEW

File: openupgrade_scripts/scripts/stock/16.0.1.1/upgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+NEW ir.actions.report: stock.action_report_picking_packages
+DEL ir.actions.report: stock.stock_replenishment_report_product_product_action
+DEL ir.actions.report: stock.stock_replenishment_report_product_template_action
+NEW ir.actions.server: stock.action_revert_inventory_adjustment
+NEW ir.model.access: stock.access_product_tag_stock_manager
+NEW ir.model.access: stock.access_stock_lot_label_layout_user
+NEW ir.model.access: stock.access_stock_lot_user
+NEW ir.model.access: stock.access_stock_picking_label_type_user
+NEW ir.model.access: stock.access_stock_replenish_option
+DEL ir.model.access: stock.access_stock_production_lot_user
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/stock/16.0.1.1/upgrade_analysis_work.txt:196]: renamed to stock.access_stock_lot_user

Comment (marielejeune) [reply openupgrade_scripts/scripts/stock/16.0.1.1/upgrade_analysis_work.txt:196]: Renamed via `rename_xmlids` in pre-migration script.

**Comment by dansanti** [comment]
works fine, thanks

File: openupgrade_scripts/scripts/stock/16.0.1.1/upgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+new model stock.replenishment.option [transient]
+# NOTHING TO DO
+
+new model stock.route
+# DONE: pre-migration: renamed model
+
+---Fields in module 'stock'---
+stock        / barcode.rule             / type (False)                  : selection_keys is now '['alias', 'expiration_date', 'location', 'location_dest', 'lot', 'pack_date', 'package', 'package_type', 'product', 'quantity', 'use_date', 'weight']' ('['alias', 'expiration_date', 'location', 'location_dest', 'lot', 'package', 'package_type', 'packaging_date', 'product', 'quantity', 'use_date', 'weight']')
+# NOTHING TO DO: no keys change in module
```
Comment (remytms) [inline openupgrade_scripts/scripts/stock/16.0.1.1/upgrade_analysis_work.txt:21]: Indeed changes that appears here are made in `barcodes_gs1_nomenclature`.

**Review by remytms — APPROVED** [review]
LGTM

**Comment by remytms** [comment]
@dansanti Can you approve this pull request? Because it seams that you tested it with success.

**Comment by andreampiovesana** [comment]
Merge?

**Review by norlinhenrik — APPROVED** [review]

**Comment by legalsylvain** [comment]
/ocabot merge nobump

thanks for this work.

**Comment by marielejeune** [comment]
> @legalsylvain your merge command was aborted due to failed check(s), which you can inspect on [this commit of 16.0-ocabot-merge-pr-4291-by-legalsylvain-bump-nobump](https://github.com/OCA/OpenUpgrade/commits/b52cb99e24ebf13c84ce438bea38e1d47d189bab).
> 
> After fixing the problem, you can re-issue a merge command. Please refrain from merging manually as it will most probably make the target branch red.

It seems that pre-commit fails on branch `16.0`.

**Comment by legalsylvain** [comment]
> It seems that pre-commit fails on branch 16.0.

Could you rebase, once https://github.com/OCA/OpenUpgrade/pull/4320 is merged ? thanks !

**Comment by marielejeune** [comment]
@legalsylvain done

**Pull request merged**